scripts/debug/split.py

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env python3
+
+import os
+import argparse
+
+argParser = argparse.ArgumentParser()
+argParser.add_argument('filename')
+
+def syntax():
+    print("Use -h for options\n")
+    exit(1)
+
+if __name__ == '__main__':
+    args = argParser.parse_args()
+    if args.filename is None:
+        syntax()
+
+    if not os.path.isfile(args.filename):
+        print(f"{args.filename} is not a file\n")
+        syntax()
+
+    ext=os.path.splitext(args.filename)[1]
+    counter = 0
+    filename = f"%03d{ext}" % counter
+    outfile = open(filename, "a")
+    with open(args.filename, "r") as file:
```

Comment:
nit: could be combined into one `with` block to avoid explicit close

Reply:
It cannot, because the outfile is closed/opened inside the loop, it's not the same outfile, it's ALL outfiles.

Reply:
Ah right, somehow missed that 👍